api/base/serializers.py

```diff
@@ -1003,7 +1009,19 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class JSONAPISerializer(ser.Serializer):
+class PrefetchRelationshipsSerializer(ser.Serializer):
+    relationship_field_names = list()
+
+    def __init__(self, *args, **kwargs):
+        super(PrefetchRelationshipsSerializer, self).__init__(*args, **kwargs)
+        for field_name, field in self.fields.iteritems():
+            if field.source == '*':
+                self.relationship_field_names.append(field_name)
```

Comment:
`relationship_field_names` is named poorly. It's the list of fields on the serializer if their source is unset then it's the field name, otherwise it's the source. Further down in the PR this list gets turned into a set and intersected with the fk fields and the m2m fields in order to get the list of relationship fields that are also on the serializer.